pipenv/project.py

```diff
@@ -856,7 +856,7 @@ def write_toml(self, data, path=None):
                 formatted_data = tomlkit.dumps(data).rstrip()
```

Comment:
What doe this `_body` check mean?

Reply:
Same as Line. 578, duck type implies tomlkit parsed result.

Reply:
Maybe it would be a good idea to extract this into a `is_tomlkit_result` function, and add a docstring to it. This folk knowledge can be easily lost.

Reply:
Also—is there a reason why tomlkit is only used to dump tomlkit-parsed results? I think it can dump normal dictionaries as well.

Reply:
Makes sense, will improve this soon.

Reply:
note that if you're dumping regular dictionaries you probably will need a way to recursively dump dictionarie ;)

Reply:
@techalchemy Yes, `tomlkit` already converts dict recursively, see the newest patch.